src/main/java/core/basesyntax/servise/strategy/impl/OperationStrategiesImpl.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.servise.strategy.impl;
+
+import core.basesyntax.servise.FruitTransaction;
+import core.basesyntax.servise.strategy.OperationService;
+import core.basesyntax.servise.strategy.OperationStrategies;
+
+public class OperationStrategiesImpl implements OperationStrategies {
+
+    @Override
+    public OperationService getOperationHandler(FruitTransaction fruitTransaction) {
+        return switch (fruitTransaction.getOperation()) {
+            case BALANCE, SUPPLY, RETURN -> new IncomingOperationService(fruitTransaction);
```

Comment:
I would recommend switching from switch expression to a hashmap

With a hashmap, adding new strategies becomes as simple as adding a new key-value pair, without needing to alter the existing decision structure. This aligns well with the open-closed principle, which states that software entities should be open for extension but closed for modification.

A hashmap can help decouple the decision logic from the execution logic. Instead of having a switch-case which is tightly coupled with all the possible cases, you can have a hashmap which dynamically retrieves the strategy based on the operation.

For a large number of cases, a hashmap can offer better performance as it provides constant-time lookups (O(1)), whereas a switch-case might degenerate to a linear search (O(n)) in the worst-case scenario